torch/csrc/Module.cpp

```diff
@@ -870,41 +831,14 @@ PyObject* initModule() {
 #ifdef USE_MLCOMPUTE
   torch::mlc::init_bindings(module);
 #endif
-  ASSERT_TRUE(THPDoubleStorage_init(module));
-  ASSERT_TRUE(THPFloatStorage_init(module));
-  ASSERT_TRUE(THPHalfStorage_init(module));
-  ASSERT_TRUE(THPLongStorage_init(module));
-  ASSERT_TRUE(THPIntStorage_init(module));
-  ASSERT_TRUE(THPShortStorage_init(module));
-  ASSERT_TRUE(THPCharStorage_init(module));
   ASSERT_TRUE(THPByteStorage_init(module));
-  ASSERT_TRUE(THPBoolStorage_init(module));
-  ASSERT_TRUE(THPQUInt8Storage_init(module));
-  ASSERT_TRUE(THPQInt8Storage_init(module));
-  ASSERT_TRUE(THPQInt32Storage_init(module));
-  ASSERT_TRUE(THPQUInt4x2Storage_init(module));
-  ASSERT_TRUE(THPBFloat16Storage_init(module));
-  ASSERT_TRUE(THPComplexDoubleStorage_init(module));
-  ASSERT_TRUE(THPComplexFloatStorage_init(module));
 
 #ifdef USE_CUDA
   // This will only initialise base classes and attach them to library namespace
   // They won't be ready for real usage until importing cuda module, that will
   // complete the process (but it defines Python classes before calling back into
   // C, so these lines have to execute first)..
-  ASSERT_TRUE(THCPDoubleStorage_init(module));
-  ASSERT_TRUE(THCPFloatStorage_init(module));
-  ASSERT_TRUE(THCPHalfStorage_init(module));
-  ASSERT_TRUE(THCPLongStorage_init(module));
-  ASSERT_TRUE(THCPIntStorage_init(module));
-  ASSERT_TRUE(THCPShortStorage_init(module));
-  ASSERT_TRUE(THCPCharStorage_init(module));
   ASSERT_TRUE(THCPByteStorage_init(module));
-  ASSERT_TRUE(THCPBoolStorage_init(module));
```

Comment:
This removal is cool, but the functions for all dtypes are still generated in torch/csrc/Storage.cpp, will be good to clean them up also (same for other functions you are removing) https://github.com/pytorch/pytorch/blob/4150b672aaa439913b66a3cf9ed4a2bae668621e/torch/csrc/Storage.cpp#L24-L45

Reply:
Yeah we should do this. I looked into it, and it seems like it will take a little bit more effort than just deleting code. For instance, removing all the `TH<type>Storage` and `THCuda<type>Storage` objects will require changing how the following functions work. It shouldn't be too complicated to do that, but maybe we should do this in a follow up PR.

https://github.com/pytorch/pytorch/blob/4150b672aaa439913b66a3cf9ed4a2bae668621e/aten/src/THC/generic/THCTensor.cpp#L366-L377

Reply:
Those functions can be deleted #65369 (the only truly used function from TH/THC is storage copy)